File: apps/ios/PagApp/PagApp/Models/RewardMock.swift
import Foundation

public enum RewardType {
    case profileScoreOnly
    case money
    case voucher
    case profileScoreAndMoney
    case profileScoreAndVoucher
}

public struct RewardResultMock {
    public let type: RewardType
    public let profileScore: Int?
    public let moneyAmount: Double?
    public let voucherInfo: String?
    
    public static let sampleMoney = RewardResultMock(type: .profileScoreAndMoney, profileScore: 50, moneyAmount: 20.0, voucherInfo: nil)
    public static let sampleVoucher = RewardResultMock(type: .profileScoreAndVoucher, profileScore: 35, moneyAmount: nil, voucherInfo: "Menü Kazandınız")
    public static let sampleProfileOnly = RewardResultMock(type: .profileScoreOnly, profileScore: 75, moneyAmount: nil, voucherInfo: nil)
}

public struct VoucherMock: Identifiable {
    public let id: String
    public let organization: String
    public let title: String
    public let status: String
    public let assignedDate: Date
    public let code: String
    
    public static let sampleList = [
        VoucherMock(id: "v-1", organization: "McDonald's", title: "Ücretsiz Big Mac Menü", status: "Kullanılabilir", assignedDate: Date(), code: "DEMO-PAG-MCD1"),
        VoucherMock(id: "v-2", organization: "Starbucks", title: "1 Adet Tall Boy Kahve", status: "Kullanıldı", assignedDate: Date().addingTimeInterval(-86400*3), code: "DEMO-PAG-STB2")
    ]
}

public struct RewardHistoryMock: Identifiable {
    public let id: String
    public let title: String
    public let amountText: String
    public let date: Date
    public let isVoucher: Bool
    
    public static let sampleList = [
        RewardHistoryMock(id: "h-1", title: "Ford Araştırması", amountText: "+₺20", date: Date(), isVoucher: false),
        RewardHistoryMock(id: "h-2", title: "PAG Genel Araştırması", amountText: "+₺50", date: Date().addingTimeInterval(-86400), isVoucher: false),
        RewardHistoryMock(id: "h-3", title: "McDonald's", amountText: "Hediye Çeki", date: Date().addingTimeInterval(-86400*2), isVoucher: true)
    ]
}

File: apps/android/app/src/main/java/com/alafteknoloji/pagapp/models/RewardMock.kt
package com.alafteknoloji.pagapp.models

import java.util.Date
import java.util.UUID

enum class RewardType {
    PROFILE_SCORE_ONLY,
    MONEY,
    VOUCHER,
    PROFILE_SCORE_AND_MONEY,
    PROFILE_SCORE_AND_VOUCHER
}

data class RewardResultMock(
    val type: RewardType,
    val profileScore: Int?,
    val moneyAmount: Double?,
    val voucherInfo: String?
) {
    companion object {
        val sampleMoney = RewardResultMock(RewardType.PROFILE_SCORE_AND_MONEY, 50, 20.0, null)
        val sampleVoucher = RewardResultMock(RewardType.PROFILE_SCORE_AND_VOUCHER, 35, null, "Menü Kazandınız")
        val sampleProfileOnly = RewardResultMock(RewardType.PROFILE_SCORE_ONLY, 75, null, null)
    }
}

data class VoucherMock(
    val id: String = UUID.randomUUID().toString(),
    val organization: String,
    val title: String,
    val status: String,
    val assignedDate: Date,
    val code: String
) {
    companion object {
        val sampleList = listOf(
            VoucherMock(id = "v-1", organization = "McDonald's", title = "Ücretsiz Big Mac Menü", status = "Kullanılabilir", assignedDate = Date(), code = "DEMO-PAG-MCD1"),
            VoucherMock(id = "v-2", organization = "Starbucks", title = "1 Adet Tall Boy Kahve", status = "Kullanıldı", assignedDate = Date(System.currentTimeMillis() - 86400 * 3000L), code = "DEMO-PAG-STB2")
        )
    }
}

data class RewardHistoryMock(
    val id: String = UUID.randomUUID().toString(),
    val title: String,
    val amountText: String,
    val date: Date,
    val isVoucher: Boolean
) {
    companion object {
        val sampleList = listOf(
            RewardHistoryMock(id = "h-1", title = "Ford Araştırması", amountText = "+₺20", date = Date(), isVoucher = false),
            RewardHistoryMock(id = "h-2", title = "PAG Genel Araştırması", amountText = "+₺50", date = Date(System.currentTimeMillis() - 86400 * 1000L), isVoucher = false),
            RewardHistoryMock(id = "h-3", title = "McDonald's", amountText = "Hediye Çeki", date = Date(System.currentTimeMillis() - 86400 * 2000L), isVoucher = true)
        )
    }
}
